### ISSUE TITLE
Persistence required on Openshift

### ISSUE THREAD

**Issue by mrethers** [mongodb/mongodb-kubernetes-operator#313]
**What did you do to encounter the bug?**
1. Install operator from OperatorHub on Openshift
2. Create a ops manager secret
3. Create a MongoDBOpsManagers instance with spec.applicationDatabase.persistence: false
4. Go to the operator log

**What did you expect?**
The database statefulset should be deployed and ready.

**What happened instead?**
Statefulset never gets into ready mode.

```
{"logType":"automation-agent-stderr","contents":"[2021/02/01 21:07:34.267] [agent.error] [components/agent.go:Iterate:171] Failed to fetch Conf"}
{"logType":"automation-agent-stderr","contents":"Unexpected conf response from server: `Response Status: 500 Internal Server Error Err: `"}
{"logType":"automation-agent-stderr","contents":"\tat monitoring-agent/components/server.go:172"}
{"logType":"automation-agent-stderr","contents":"\tat monitoring-agent/components/server.go:219"}
{"logType":"automation-agent-stderr","contents":"\tat monitoring-agent/components/agent.go:169"}
{"logType":"automation-agent-stderr","contents":"\tat monitoring-agent/components/agent.go:261"}
{"logType":"automation-agent-stderr","contents":"\tat monitoring-agent/monitoringexecutor/executor.go:125"}
{"logType":"automation-agent-stderr","contents":"\tat louisaberger/procexec/concurrency.go:45"}
{"logType":"automation-agent-stderr","contents":"\tat src/runtime/asm_amd64.s:1357"}
{"logType":"automation-agent-stderr","contents":"[2021/02/01 21:07:34.267] [agent.info] [components/agent.go:Run:266] Done. Sleeping for 5s..."}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:34.628] [.info] [main/components/agent.go:LoadClusterConfig:197] [21:07:34.628] clusterConfig unchanged"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:35.628] [.info] [main/components/agent.go:LoadClusterConfig:197] [21:07:35.628] clusterConfig unchanged"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.042] [.info] [cm/mongoclientservice/mongoclientservice.go:sendClientToRequesters:599] [21:07:36.042] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down.  Informing all client requests and disposing of client (0x0)"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [cm/mongoctl/processctl.go:RunCommand:934] <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [state/stateutil/fickleutil.go:currentProcessState:225] <ops-manager-db-1> [21:07:36.043] Error getting pid : <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [state/stateutil/stateutil.go:ComputeCurrentState:89] <ops-manager-db-1> [21:07:36.043] Error getting fickle state for current state : <ops-manager-db-1> [21:07:36.043] Error getting pid : <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [cm/director/director.go:updateCurrentState:733] <ops-manager-db-1> [21:07:36.043] Error calling ComputeState : <ops-manager-db-1> [21:07:36.043] Error getting fickle state for current state : <ops-manager-db-1> [21:07:36.043] Error getting pid : <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [cm/director/director.go:planAndExecute:504] <ops-manager-db-1> [21:07:36.043] Failed to compute states : <ops-manager-db-1> [21:07:36.043] Error calling ComputeState : <ops-manager-db-1> [21:07:36.043] Error getting fickle state for current state : <ops-manager-db-1> [21:07:36.043] Error getting pid : <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
{"logType":"automation-agent-verbose","contents":"[2021/02/01 21:07:36.043] [.error] [cm/director/director.go:mainLoop:379] <ops-manager-db-1> [21:07:36.043] Failed to planAndExecute : <ops-manager-db-1> [21:07:36.043] Failed to compute states : <ops-manager-db-1> [21:07:36.043] Error calling ComputeState : <ops-manager-db-1> [21:07:36.043] Error getting fickle state for current state : <ops-manager-db-1> [21:07:36.043] Error getting pid : <ops-manager-db-1> [21:07:36.043] Server at ops-manager-db-1.ops-manager-db-svc.handshake.svc.cluster.local:27017 (local=false) is down"}
```

**Operator Information**
 - 1.9.0
 - 
**Kubernetes Cluster Information**
 Openshift 4.5

**Additional context**
This goes away if I remove that persistence line and let the Statefulsets create the PVCs.

```apiVersion: mongodb.com/v1
kind: MongoDBOpsManager
metadata:
  name: ops-manager
  namespace: handshake
spec:
  adminCredentials: mongodb-ops-manager-credentials
  applicationDatabase:
    members: 3
    persistent: false
  configuration:
    mms.fromEmailAddr: admin@thecompany.com
  externalConnectivity:
    type: LoadBalancer
  replicas: 1
  version: 4.4.7
status:
  applicationDatabase:
    lastTransition: '2021-02-01T21:00:30Z'
    members: 3
    observedGeneration: 1
    phase: Running
    version: 4.2.2-ent
  opsManager:
    lastTransition: '2021-02-01T21:00:30Z'
    message: StatefulSet not ready
    observedGeneration: 1
    phase: Pending
    resourcesNotReady:
      - kind: StatefulSet
        message: 'Not all the Pods are ready (total: 1, updated: 1, ready: 0)'
        name: ops-manager
    url: 'http://ops-manager-svc.handshake.svc.cluster.local:8080'
```

**Comment by chatton** [comment]
Hi @mrethers 

This was a problem with the 1.9.0 release which should be resolved with the 1.9.1 release. Please let me know if the problem persists.

(Note: this repo is the Community Operator repo, if there are still problems could you create a new issue against the [enterprise repo](https://github.com/mongodb/mongodb-enterprise-kubernetes))

**Comment by jamesbroadhead** [comment]
This ticket relates to the Enterprise Operator, and this repo is for the Community Operator. 
For support on the Enterprise Operator, you can use the [Enterprise Repo](https://github.com/mongodb/mongodb-enterprise-kubernetes), refer to the [Enterprise Operator Docs](https://docs.mongodb.com/kubernetes-operator/master/) visit the [Support Center](https://support.mongodb.com/welcome). 
Hope that these links help you resolve your issue
